### PR TITLE
refactor: set enable_experimental_rbac_check default 1

### PR DESCRIPTION
**Pull request by TCeason**
I hereby agree to the terms of the CLA available at: https://docs.databend.com/dev/policies/cla/

## Summary

set enable_experimental_rbac_check default 1

In the default setting of enable_experimental_rbac_check

User can drop udf/stage that is owned by themself role.

Note:

Because of the flag `if exists`, need to get udf/stage in privilege_access.rs .

- Fixes #[Link the issue here]

## Tests

- [ ] Unit Test
- [x] Logic Test
- [ ] Benchmark Test
- [ ] No Test  - _Explain why_

## Type of change

- [ ] Bug Fix (non-breaking change which fixes an issue)
- [ ] New Feature (non-breaking change which adds functionality)
- [ ] Breaking Change (fix or feature that could cause existing functionality not to work as expected)
- [ ] Documentation Update
- [x] Refactoring
- [ ] Performance Improvement
- [ ] Other (please describe):

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/15436)

### CODE REVIEW

File: src/query/service/src/interpreters/access/privilege_access.rs [hunk 1]
```diff
@@ -618,13 +619,49 @@ impl AccessChecker for PrivilegeAccess {
...
+                if enable_experimental_rbac_check {
+                    let udf = HashSet::from([udf_name]);
+                    self.validate_udf_access(udf).await?;
+                } else {
+                    self.validate_access(&GrantObject::Global, UserPrivilegeType::Drop)
+                                        .await?;
+                }
+            }
+            Plan::DropStage(plan) => {
+                match UserApiProvider::instance().get_stage(&tenant, &plan.name).await {
```
Comment (TCeason) [inline src/query/service/src/interpreters/access/privilege_access.rs:642]: The user stage or tempory stage can get by this API? cc @flaneur2020 @youngsofun

Comment (flaneur2020) [reply src/query/service/src/interpreters/access/privilege_access.rs:642]: if i understands correctly, the user stage is not allowed to be dropped

**Comment by flaneur2020** [comment]
If there are some legacy stages, they would have some risks not being able to access that stage after this PR gets merged.

However, I think this change is necessary. We can have a standby with potential customer feedbacks and providing solutions to move forward on it.

**Review by flaneur2020 — APPROVED** [review]

File: src/query/service/src/interpreters/access/privilege_access.rs (hunk 1 above)
Comment (youngsofun) [reply src/query/service/src/interpreters/access/privilege_access.rs:642]:
```
root@localhost:8000/default> drop stage ~;
error: APIError: ResponseError with 2506: user stage is not allowed to be dropped

```

Comment (youngsofun) [reply src/query/service/src/interpreters/access/privilege_access.rs:642]: > The user stage or tempory stage can get by this API? cc @flaneur2020 @youngsofun

yes, I think

Comment (youngsofun) [reply src/query/service/src/interpreters/access/privilege_access.rs:642]: what is `tempory stage`?

**Comment by youngsofun** [comment]
> If there are some legacy stages, they would have some risks not being able to access that stage after this PR gets merged.
> 
> However, I think this change is necessary. We can have a standby with potential customer feedbacks and providing solutions to move forward on it.

this pr only affect drop, for legacy stages, user can  set=enable_experimental_rbac_check=0 and then drop?

File: src/query/service/src/interpreters/access/privilege_access.rs (hunk 1 above)
Comment (TCeason) [reply src/query/service/src/interpreters/access/privilege_access.rs:642]: > what is `tempory stage`?

is_temporary = true;

pub struct StageInfo {
    pub stage_name: String,
    pub stage_type: StageType,
    pub stage_params: StageParams,
    // on `COPY INTO xx FROM 's3://xxx?ak=?&sk=?'`, the URL(ExternalLocation) will be treated as an temporary stage.
    pub is_temporary: bool,

**Review by youngsofun — APPROVED** [review]

File: src/query/service/src/interpreters/access/privilege_access.rs (hunk 1 above)
Comment (youngsofun) [reply src/query/service/src/interpreters/access/privilege_access.rs:642]: @TCeason  no, `tempory stage` are not stored in meta server, and not accessible outside .

Comment (TCeason) [reply src/query/service/src/interpreters/access/privilege_access.rs:642]: > what is `tempory stage`?

is_temporary = true;

pub struct StageInfo {
    pub stage_name: String,
    pub stage_type: StageType,
    pub stage_params: StageParams,
    // on `COPY INTO xx FROM 's3://xxx?ak=?&sk=?'`, the URL(ExternalLocation) will be treated as an temporary stage.
    pub is_temporary: bool,

**Comment by youngsofun** [comment]
need to update the tests? @TCeason

**Comment by TCeason** [comment]
> this pr only affect drop, for legacy stages, user can set=enable_experimental_rbac_check=0 and then drop?

Yes. Or grant ownership to user owned role

**Pull request merged**